decentraland/unity-explorer
# chore: refactor tween system (Mirror scene)

## What does this PR change?

 Refactor TweenUpdaterSystem:

1. TweenSystem now modifies a `ICustomTweener` instead of the Transform in `TarnsformComponent`. Then it writes the result in `SDKTransform`, which then updates the `TransformComponent` in `UpdateTransformSystem`. In this way, every system is in correct order and sync.
2. A pool was introduces for `ICustomTweener` and its reusage
3. Added missing parenting for transform component when writing back to CRDT
4. Removed `SDKTweenModel` to avoid redudancy. Had to keep a cached version of PBTween though, because apparently there are edge cases where the scene does not set PBTween as dirty when it should.
5. Avoids writing over and over the result to CRDT by prober state checking of the tween


## How to test the changes?

Recomendation: In Debug Menu, in Landscape setting, move the load raidus to 1.

1. Go to Goerli
2. Go to the mirror scene (82,-5). You should be able to play as in explorer (complete the puzzle). Do this puzzle in first person mode; third person mode doesnt have a ray thats long enough. Also, use mouse lock. Its going to make it easier
3. Go to the moving platforms scene (77,-5). You should be able to play as in explorer

## Our Code Review Standards

https://github.com/decentraland/unity-renderer/blob/master/docs/code-review-standards.md
